internal/campaigns/types.go

```diff
@@ -727,11 +734,44 @@ func ComputeReviewState(c *Changeset, events ChangesetEvents) (ChangesetReviewSt
 	return events.reviewState()
 }
 
-func computeBitbucketBuildStatus(pr *bitbucketserver.PullRequest) ChangesetCheckState {
-	var states []ChangesetCheckState
-	for _, status := range pr.BuildStatuses {
-		states = append(states, parseBitbucketBuildState(status.State))
+func computeBitbucketBuildStatus(lastSynced time.Time, pr *bitbucketserver.PullRequest, events []*ChangesetEvent) ChangesetCheckState {
+	var latestCommit bitbucketserver.Commit
+	for _, c := range pr.Commits {
+		if latestCommit.CommitterTimestamp < c.CommitterTimestamp {
+			latestCommit = *c
+		}
+	}
+
+	stateMap := make(map[string]ChangesetCheckState)
+
+	// States from last sync
+	for _, status := range pr.CommitStatus {
+		if status.Commit != latestCommit.ID {
+			continue
+		}
+		stateMap[status.Key()] = parseBitbucketBuildState(status.Status.State)
+	}
+
+	// Add any we've received via webhooks since our last sync
+	for _, e := range events {
+		switch m := e.Metadata.(type) {
+		case *bitbucketserver.CommitStatus:
+			if m.Commit != latestCommit.ID {
+				continue
+			}
+			dateAdded := unixMilliToTime(m.Status.DateAdded)
+			if !dateAdded.After(lastSynced) {
```

Comment:
How about this:

```suggestion
			if dateAdded.Before(lastSynced) {
```

If there was an event at the exact same time we did a sync, we don't want to ignore it